src/main/kotlin/nl/avisi/structurizr/site/generatr/site/views/SoftwareSystemComponentPage.kt

```diff
@@ -5,6 +5,9 @@ import nl.avisi.structurizr.site.generatr.site.model.SoftwareSystemComponentPage
 
 fun HTML.softwareSystemComponentPage(viewModel: SoftwareSystemComponentPageViewModel) {
     softwareSystemPage(viewModel) {
+        if (viewModel.diagrams.none())
```

Comment:
May be it is more clean to introduce `viewModel.isHidden` or similar here and for the other occasions. That would also open the possibility for simple tests.